applications/rag/example_notebooks/rag-ray-ingest-with-kubernetes-docs.ipynb

```diff
@@ -0,0 +1,285 @@
+     "This is a Python notebook for generating the vector embeddings based on [Kubernetes docs](https://github.com/dohsimpson/kubernetes-doc-pdf/) used by the RAG on GKE application.   \n",
+     "For full information, please checkout the GitHub documentation [here](https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/applications/rag/README.md).\n",
+     "\n",
+     "\n",
+     "\n",
+     "## Clone the kubernetes docs repo"
+    ]
+   },
+   {
+    "cell_type": "code",
+    "execution_count": null,
+    "id": "5f9b1fad-537e-425f-a5fc-587a408b1fab",
+    "metadata": {},
+    "outputs": [],
+    "source": [
+     "!mkdir /data/kubernetes-docs -p\n",
+     "!git clone https://github.com/dohsimpson/kubernetes-doc-pdf /data/kubernetes-docs\n"
+    ]
+   },
+   {
+    "cell_type": "markdown",
+    "id": "b984429c-b65a-47b7-9723-ee3ad81d61db",
+    "metadata": {},
+    "source": [
+     "## Install the required packages"
+    ]
+   },
+   {
+    "cell_type": "code",
+    "execution_count": null,
+    "id": "40e4d29d-79c6-4233-a8ed-0f8a42576656",
+    "metadata": {
+     "scrolled": true
+    },
+    "outputs": [],
+    "source": [
+     "!pip install langchain langchain-community sentence_transformers pypdf"
+    ]
+   },
+   {
+    "cell_type": "markdown",
+    "id": "f80cc5af-a1fa-456d-a4ed-fa2ffa3b87a0",
+    "metadata": {},
+    "source": [
+     "## Writting job to be used on the Ray Cluster"
+    ]
+   },
+   {
+    "cell_type": "code",
+    "execution_count": null,
+    "id": "36523f3f-0c93-41da-abb9-c113bb456bc1",
+    "metadata": {},
+    "outputs": [],
+    "source": [
+     "# Create a directory to package the contents that need to be downloaded in ray worker\n",
+     "! mkdir -p rag-app"
+    ]
+   },
+   {
+    "cell_type": "code",
+    "execution_count": null,
+    "id": "69d912e5-2225-4b44-80cd-651f7cc71a40",
+    "metadata": {},
+    "outputs": [],
+    "source": [
+     "%%writefile rag-app/job.py\n",
+     "\n",
+     "import os\n",
+     "import uuid\n",
+     "import glob\n",
+     "\n",
+     "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
+     "from langchain.embeddings import HuggingFaceEmbeddings\n",
+     "from langchain_community.document_loaders import PyPDFLoader\n",
+     "\n",
+     "from google.cloud.sql.connector import Connector, IPTypes\n",
+     "import sqlalchemy\n",
+     "\n",
+     "from sqlalchemy.ext.declarative import declarative_base\n",
+     "from sqlalchemy import Column, String, Text, text, JSON\n",
+     "from sqlalchemy.orm import scoped_session, sessionmaker, mapped_column\n",
+     "from pgvector.sqlalchemy import Vector\n",
+     "\n",
+     "# initialize parameters\n",
+     "\n",
+     "INSTANCE_CONNECTION_NAME = os.environ[\"CLOUDSQL_INSTANCE_CONNECTION_NAME\"]\n",
+     "print(f\"Your instance connection name is: {INSTANCE_CONNECTION_NAME}\")\n",
+     "VECTOR_EMBEDDINGS_TABLE_NAME = os.environ.get(\"EMBEDDINGS_TABLE_NAME\", \"rag_vector_embeddings\")\n",
```

Comment:
I encountered error `relation "rag_embeddings_db" does not exist` when submitting prompt via Frontend UI, see below. I think it's because the `EMBEDDINGS_TABLE_NAME` env is not set correctly, so here the table is created with name `rag_vector_embeddings` rather than `rag_embeddings_db`